### ISSUE TITLE
Can't access property 'item' in dynamic object

### ISSUE THREAD

**Issue by RobWalker** [sebastienros/jint#885]
This looks to be related to #642 but isn't solved but that fix.

The following test will fail:
```
        [Fact]
        public void CanAccessMemberNamedItemThroughExpando()
        {
            var parent = (IDictionary<String, Object>)new ExpandoObject();
            var child = (IDictionary<String, Object>)new ExpandoObject();
            var values = (IDictionary<String, Object>)new ExpandoObject();

            parent["child"] = child;
            child["item"] = values;
            values["title"] = "abc";

            _engine.SetValue("parent", parent);
            Assert.Equal("abc", _engine.Execute("parent.child.item.title").GetCompletionValue());
        }
```

Replacing the name `item` with `zzitem` (in both places) allows the test to pass.

Thanks.

**Comment by lahma** [comment]
Thanks for the report, fixed in #886 .

**Comment by RobWalker** [comment]
Thanks for the amazingly fast response!
